sw/device/silicon_creator/rom/base_rom_start.S

```diff
@@ -0,0 +1,443 @@
+  // of `mie`.
+  li   t0, 0x00000888
+  csrc mie, t0
+
+  // Check if AST initialization should be skipped.
+  li   a0, (TOP_DARJEELING_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET)
+  lw   t0, OTP_CTRL_PARAM_CREATOR_SW_CFG_AST_INIT_EN_OFFSET(a0)
+  li   t1, MULTIBIT_ASM_BOOL4_TRUE
+  bne  t0, t1, .L_ast_init_end
+
+  // Copy the AST configuration from OTP.
+  li   a0, (TOP_DARJEELING_AST_BASE_ADDR)
+  li   a1, (TOP_DARJEELING_AST_BASE_ADDR + AST_REGAL_REG_OFFSET + 4)
+  li   a2, (TOP_DARJEELING_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET + \
+            OTP_CTRL_PARAM_CREATOR_SW_CFG_AST_CFG_OFFSET)
+  call crt_section_copy
+
+  // Enable jittery clock if enabled in OTP.
+  li   a0, (TOP_DARJEELING_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET)
+  lw   t0, OTP_CTRL_PARAM_CREATOR_SW_CFG_JITTER_EN_OFFSET(a0)
+  li   a0, TOP_DARJEELING_CLKMGR_AON_BASE_ADDR
+  sw   t0, CLKMGR_JITTER_ENABLE_REG_OFFSET(a0)
+
+.L_ast_init_end:
+  // Check if we are trying to transition to RMA. If so, busy loop for
+  // CREATOR_SW_CFG_RMA_SPIN_CYCLES to let the transition start and continue
+  // looping while lc_ctrl is not ready. Reset if the CPU is still executing at
+  // the end.
+  li   a0, (TOP_DARJEELING_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET)
+  lw   t0, OTP_CTRL_PARAM_CREATOR_SW_CFG_RMA_SPIN_EN_OFFSET(a0)
+  li   t6, HARDENED_BOOL_TRUE
+  bne  t0, t6, .L_entropy_enable
+
+  li   a1, (TOP_DARJEELING_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET)
+  lw   t1, OTP_CTRL_PARAM_CREATOR_SW_CFG_RMA_SPIN_EN_OFFSET(a1)
+  beq  t1, t6, .L_rma_spin_check_straps
+  unimp
+  unimp
+  unimp
+  unimp
+
+.L_rma_spin_check_straps:
+  // Configure strap pins.
+  //
+  // | GPIO Pin:        | 24             | 23             | 22               |
+  // |                  | (SW_STRAP_2)   | (SW_STRAP_1)   | (SW_STRAP_0)     |
+  // |------------------+----------------+----------------+------------------|
+  // | RMA Entry Value: | Strong pull-up | Strong pull-up | Strong pull-down |
+  // |------------------+----------------+----------------+------------------|
+  // | Configuration:   | Input          | Input          | Input            |
+  // |                  | Pull-down      | Pull-down      | Pull-up          |
+  li   a0, (TOP_DARJEELING_PINMUX_AON_BASE_ADDR + \
+            PINMUX_MIO_PAD_ATTR_0_REG_OFFSET)
+  li   t0, (1 << PINMUX_MIO_PAD_ATTR_0_PULL_EN_0_BIT) | \
+           (1 << PINMUX_MIO_PAD_ATTR_0_PULL_SELECT_0_BIT)
+  sw   t0, (SW_STRAP_0_PAD * 4)(a0)
+  li   t0, (1 << PINMUX_MIO_PAD_ATTR_0_PULL_EN_0_BIT)
+  sw   t0, (SW_STRAP_1_PAD * 4)(a0)
+  sw   t0, (SW_STRAP_2_PAD * 4)(a0)
+  li   a0, (TOP_DARJEELING_PINMUX_AON_BASE_ADDR + \
+            PINMUX_MIO_PERIPH_INSEL_0_REG_OFFSET)
+
+  // Spin for a minimum of 5us worth of cycles (assuming 100MHz clock), while we
```

Comment:
The above comments for the watchdog threshold say "assuming a 62.5 MHz clock"; so these are different clock domains? Might be good to briefly mention which one in both places to avoid confusion